New UI

claudey opened the pull request:
@Shpigford I saw the screenshots on Twitter and decided to take an early stub. Would you share the Figma files? 🤞🏽
![maybe-new-ui-stub](https://github.com/user-attachments/assets/f729d76e-235a-4f4c-9fb6-a30f282956fe)

claudey committed:
comment Vernier for now. Was not install on local.

claudey committed:
bin/setup

claudey committed:
Removed the global width boundary.

claudey committed:
Added the "Transactions handled your way" section, except for images.

claudey committed:
added images for the transparency section.

claudey committed:
Moved footer constraints

claudey committed:
Added transparency section.

claudey committed:
More images, border strip, and github contrib image.

claudey committed:
Removed images from previous UI

claudey committed:
NavHelper to keep footer links in one place.

claudey committed:
Add legal section

claudey committed:
Add footer border and changed text.

claudey committed:
Used NavHelper footer_resources object.

claudey committed:
Added top border strip.

claudey committed:
Update footer.

claudey committed:
Updated navbar with fewer links and sign-in

claudey committed:
Added an object for footer tools.

claudey committed:
Replaced with loop block of tooter tools.

claudey committed:
Added more links to NavHelper

claudey committed:
Extracted footer content into a partial.

claudey committed:
Footer partial in use.

claudey committed:
If number is nil, multiply by 0.

claudey committed:
Formatting changes.

claudey commented:
Updated the footer.
![Screenshot 2025-03-21 at 10 40 33 AM](https://github.com/user-attachments/assets/12cd8277-d911-4aa1-95a2-9244a90ba884)

Shpigford commented:
@claudey This is amazing! Here's the Figma file.

https://www.figma.com/design/dftxSj2ZnDRMWwE9pP4um7/%F0%9F%9F%A3-%5BMAYBE-90%5D--Marketing-Site-Update?node-id=1-457&m=dev

If you want to get it to a point where I can do a code/quality check on it, I'm happy to set up a bounty on this for you.

claudey commented:
Yes. I'd like that. Go ahead.

claudey committed:
Initial pages for features.

claudey committed:
Added a features controller.

claudey committed:
routes for features.

claudey committed:
Added charts for first section on Tracking page

claudey committed:
Base text for Sections on tracking page.

claudey committed:
Imported Geist and Geist Mono font via Google fonts, for now.

claudey committed:
Added extensions to use Geist fonts inline.

claudey committed:
Completed one sub-feature of the tracking.

claudey committed:
Add svg images for tracking page.

claudey committed:
Completed tracking page feature styling and content.

claudey committed:
Added variations of the background chart.

claudey committed:
Added header image for Tracking page.

claudey committed:
Wrapped page in an article so i can use the background on it. Background image showing now.

claudey committed:
Added a background color to main to match the cutoff color of the main background illustration.

claudey committed:
exported more space under main bg illustration so there's no lurking gradient.

claudey committed:
Added styling for header areas for tracking page.

claudey committed:
Added the gradient under the waitlist form.

claudey committed:
All on header text Geist.

claudey committed:
Formatting.

claudey committed:
Styled form to match Figma.

claudey commented:
![maybe-features-tracking](https://github.com/user-attachments/assets/ebd06210-2ef7-452f-b865-e6777a78feee)

claudey committed:
Added white bordered area to main text.

claudey committed:
Add images for Transactions feature page.

claudey committed:
Fix typo.

claudey committed:
rename image.

claudey committed:
Update to make image full width but with a max of 600px.

claudey committed:
Transactions feature page done.

claudey committed:
Add images for budgeting feature page.

claudey committed:
Budgeting feature page done.

claudey commented:
![maybe-features-budgeting](https://github.com/user-attachments/assets/cd7ce775-c3eb-4934-9b6d-d122e017d8d2)

claudey committed:
Fixed footer to match Figma style.

claudey committed:
Exported actual footer bg.

claudey committed:
Found the main site baackgorund which runs through all pages.

claudey committed:
Replaced Inter Var with Geist as default.

claudey committed:
No need for "font-geist" class.

claudey committed:
Placed site background.

claudey committed:
Updated Financial terms page to match Figma.

claudey committed:
Some space atop the footer.

claudey committed:
Removed the bg-color on the footer.

claudey committed:
Replaced top image with svg

claudey committed:
Gave the terms page a container

claudey committed:
Added a filter to list Tools by category.

claudey committed:
Completed filter styling for Tools page.

claudey committed:
Completed styling Tools page.

claudey committed:
Fixed spacing under categories strip.

claudey commented:
![maybe-tools-page](https://github.com/user-attachments/assets/3e500e09-964e-47c0-a685-a7e56c7b9639)

claudey committed:
Removed the min-height param since tools list is stacked now.

claudey committed:
Validate categories and sanitize characters in query

claudey committed:
apostrophe fix.

claudey committed:
Added svg assets for terms show page.

claudey committed:
Implemented copy button.

claudey committed:
Formatted Terms show page to match Figma.

claudey commented:
![Screenshot 2025-03-24 at 9 34 38 AM](https://github.com/user-attachments/assets/455f9f26-2126-4c9f-94e1-93a7428820c9)

claudey committed:
Extracted share buttons into a partial since it's used on the tool show page as well.

claudey committed:
Added scope for latest and published.

claudey committed:
Pagination for articles list.

claudey committed:
Blog index page structure done, left with styling

claudey committed:
Remove waitlist partial from under each blog article.

Shpigford commented:
@claudey Thanks so much for the continued work on this! Going to be reviewing work thus far on it. Anything I should keep in mind while I'm checking it out (known issues, etc)?

claudey committed:
Updated the articles/index to match figma.

claudey commented:
@Shpigford I've completed the desktop views for the following pages

## Done
- [x] Features > Track
- [x] Features > Transactions
- [x] Features > Budget
- [x] Blog index
- [x] Financial terms > Index
- [x] Financial terms > show

## Up next

- [ ] Menu
- [ ] home page
- [ ] Stocks pages
- [ ] Adjusted mobile views for all pages.

claudey committed:
Updated blog show page to match Figma.

claudey commented:
![maybe-blog-show](https://github.com/user-attachments/assets/a0399899-4cd9-4f56-afe7-5aefea9b8b42)

claudey committed:
Added subtle gradient on articles index.

claudey committed:
Renamaed geist-mono to mono since all font is Geist.

claudey committed:
Updated the main gradient bg.

claudey committed:
Fixed the transparent-to-white background gradient on the main content on the Features pages.

claudey committed:
Updated icons and text for features pages.

claudey committed:
Removed duplicate waitlist form from page.

claudey committed:
Center strip with categories.

claudey committed:
Added a variable to show More Tools section

claudey committed:
Fixed Tools/Show page to match Figma.

claudey committed:
Copied the random_sample function over to use in the "More tools" section.

claudey committed:
Added "Input your details" as per Figma.

claudey committed:
Updated the submit button.

claudey committed:
Add some icons and assets for home page.

claudey committed:
Home page started, but committing WIP.

claudey committed:
Had "tracking-0" which was causing tailwind to break. All fixed now.

claudey committed:
vector Images for home page

claudey committed:
Home page matches Figma. Jsutin gave me a run!

claudey committed:
Formatting

claudey committed:
Fixed background on github and discord badges.

claudey committed:
Images for nav sections.

claudey committed:
Added more objects to navhelper

claudey committed:
Removed empty link to "Featrues" at the top of the page.

claudey committed:
Extracted navbar into partial.

claudey committed:
Extracted resources mega_menu

claudey committed:
Extracted navbar here. Site nav menu working now.

claudey commented:
@Shpigford update

- [x] Home page content
- [x] nav menu

## Finish Line
- [ ] AI section on home page
- [ ] stocks pages
- [ ] mobile views for all pages.

![maybe-home-page](https://github.com/user-attachments/assets/a2b23396-7e38-499b-bab4-1df1e19a15bc)

claudey committed:
Replaced bottom background on footer with inline image.

claudey committed:
Added background blur, brought the border strip into the navbar and made navbar sticky.

claudey committed:
Added footer icons.

claudey committed:
Styled footer to match Figma.

claudey committed:
Added an override to restyle the pagigation section.

claudey committed:
Moved header nav links up and footer to the bottom.

claudey committed:
Added glowing logo in SVG and PNG

claudey committed:
Added glowing logo and styled as in Figma.

claudey committed:
Moved border strip into navbar partial.

claudey committed:
Added glowing logo on articles/index too.

claudey committed:
Top padding on home page to match Figma.

claudey committed:
Updated waitlist partial to use glowing logo.

Shpigford commented:
It's looking great! Thank you so much!

claudey committed:
Included the pluralize helper to correctly state "stock" or "stocks" in search results.

claudey committed:
Implemented the "no results found" block for the search results.

claudey committed:
Pluralize

claudey committed:
Styling fix.

claudey committed:
Created a list of "featured" stocks and exchanges to select from.

claudey committed:
Added a faetured_stocks variable which randomly selects 12 out of the manually entered "FEATURED_STOCKS" variable.

claudey committed:
Crated an object with manually selected exchanges which can be used on the stocks index page.

claudey committed:
Also created a function that returned the stocks, exchanges and sectors that will be shown under the search form

claudey committed:
Extracted the quick links bar into a partial.

claudey committed:
"Related" to "Other"

claudey committed:
Remove margin bottom, import quick links bar

claudey committed:
Put page content into an <article>.

claudey committed:
Fix up the "Sectors" section to select random sectors to show.

claudey committed:
Stock exchanges section uses list from StockHelper

claudey committed:
Stocks section also uses 12 random stocks from StockHelper.

claudey committed:
Render "quick_links" and formatting.

claudey committed:
Pass @total_stocks so the count it shown in the UI.

claudey committed:
Page updates from Justin via Figma comments.

claudey committed:
Show stock, and format "Powered by synth" badge.

claudey committed:
Increased border-radius for stock to USD converter.

claudey committed:
Revert sort param to earlier value.

claudey committed:
Created new views for exchanges and sectors. I feel these might have to be removed later.

claudey committed:
Formatting.

claudey committed:
Added a nil check when building the @exchanges variable.

claudey committed:
Merge branch 'main' into stub-new-ui

claudey commented:
@Shpigford Stocks pages are done. I'm getting some feedback from Justin on a few changes.

![maybe-stocks-page](https://github.com/user-attachments/assets/16f70c2a-4040-4051-a17f-401d2bfacbf3)

Shpigford commented:
@claudey Anything you need from us to get this to the finish line?

claudey commented:
@shpigford @justinfar mentioned he was making a few changes since the launch date has been moved. I'll reach out to him and share an update when he replies.

claudey committed:
Extracted cards into YML for local parsing.

claudey committed:
Updated the features controller to handle the page content.

claudey committed:
Created partials to load each tab

claudey committed:
Base card to show questions and answers

claudey committed:
Added a controller for the tabs.

claudey committed:
Partial to load the tab titles and their icons.

claudey committed:
Created this to hold the tab data and to load the tab content.

claudey committed:
Added assistant link to first menu item on nav.

claudey committed:
Installed masonry-layout to handle the masonry content.

claudey committed:
Added the page for /features/assistant

claudey committed:
A masonry controller.

claudey committed:
Added the assistant route.

claudey committed:
Add Masonry to importmap

claudey committed:
Merge branch 'maybe-finance:main' into stub-new-ui

claudey committed:
Added avatar for assistant in png.

claudey committed:
Styled cards for assistant tabs.

claudey commented:
![Screenshot 2025-04-17 at 5 21 51 AM](https://github.com/user-attachments/assets/b09aa2a2-8afc-45c7-8cb7-c179257b97a7)

claudey committed:
Replaced the home hero image.

claudey committed:
Updated the home hero image.

claudey committed:
Fixed the tabs on the Assistant page to match the design.

claudey committed:
Remove the individual file call for the separate tabs and load inline.

claudey committed:
Update Assistant page content.

claudey committed:
Update card code to match Figma.

claudey committed:
Fixed the class toggle on active tabs.

claudey committed:
Added the image for the header on the Assistant page.

claudey committed:
Add images for assistant page.

claudey committed:
Updated content for Assistant page.

claudey committed:
Merge branch 'main' into stub-new-ui

claudey commented:
![maybe-assistant-page](https://github.com/user-attachments/assets/12280de6-59c6-437d-b109-f0ebe80d250e)

claudey committed:
Redirect /ai to /features/assistant.

Shpigford commented:
@claudey Looking really great! Think we can wrap it up in the next couple of days?

claudey commented:
@Shpigford Yes, Josh. We definitely can. I'll keep pushing with updates.

claudey committed:
Merge branch 'main' into stub-new-ui

claudey committed:
Added the pricing page. Tier cards added now.

claudey committed:
Created a controller to handle the duration toggle [monthly/yearly]

claudey committed:
Also generated a controller for handling FAQ section.

claudey committed:
Merge remote-tracking branch 'refs/remotes/claudey/stub-new-ui' into stub-new-ui

claudey committed:
Added pricing to navbar and a current_page check to highlight title.

claudey committed:
Added pricing to routes.

claudey committed:
run new migrations.

claudey committed:
Extracted FAQs to a helper file for leaner erb and modular questions.

claudey committed:
F`ixed the FAQs

claudey committed:
Uploaded meta in both png and jpg.

claudey committed:
Replaced old meta image with new jpg one.

claudey committed:
Added a boxShadow extention for to use on images.

claudey committed:
Used shadow sofy on Maybe+ card on pricing page. Works great!

claudey committed:
Added a small helper function to round the number of stars.

claudey committed:
Now showing star count form api.

claudey committed:
Added shadowo effect for buttons.

claudey committed:
Created a new cta_header to replace previous one.

claudey committed:
Updated buttons on navbar to match.

claudey committed:
Assets for mobile menu

claudey committed:
Changed to new cta_header

claudey committed:
Crated a mobile version of the resources menu.

claudey committed:
Added a fallback for a hard-coded number in case star count API is unavailable.

claudey committed:
Hnadle dropdowns.

claudey committed:
Created a mobile menu, like in Figma.

claudey committed:
Navbar controller, for navbar interactions.

claudey committed:
Add styles for mobile menu.

claudey committed:
Added assets for AI section on home page.

claudey committed:
Updated the images from Figma.

claudey committed:
Put AI Section in a partial for now.

claudey committed:
AI section mostly done, except gradients, etc.

claudey committed:
Added assets for pricing page.

claudey committed:
Used some backgrounds and images in place of HTML-codded gradients.

claudey committed:
Added the linear fill on the price as a background and clipped it on the text.

claudey commented:
@Shpigford I'm converting this into a full PR because the major page changes have been done, including the mobile navigation. The design details are so many so I've had to skip a few for now.

While you review, I'll take today to make all the pages fit mobile better.

claudey committed:
Added the stylized Maybe-plus icon at the top.

Shpigford commented:
@claudey Sounds good!

claudey committed:
Footer all responsive now.

claudey committed:
Fixed image sizing for mobile screens; no overflowing image now.

claudey committed:
Fixed padding on navbar.

claudey committed:
Made a working breakthrough for header images.

Export from Figma without the shadow effect, then apply them with CSS. Works perfectly!

claudey committed:
Home section hero images replaced.

claudey committed:
Added a new class to style the prompt input image in the ai section.

claudey committed:
Add effects.

claudey committed:
Add effects to hero images.

claudey committed:
Add left-right padding to footer till 1280px.

claudey committed:
Replaced header for Tracking page.

claudey committed:
Tracking page repsonsive now.

claudey committed:
Replaced transactions header.

claudey committed:
Transactions page fully responsive now.

claudey committed:
Replaced budgeting header.

claudey committed:
Budgeting page responsive now.

claudey committed:
Extract AI features into a helper.

claudey committed:
Assistant page responsive now.

claudey committed:
Fixed responsiveness on ai conversation card too.

claudey committed:
More padding on Transactions page.

claudey committed:
Fixed reponsiveness on AI conversation cards.

claudey committed:
Used the box-shadow for the assitant cards.

claudey committed:
Format.

claudey committed:
Revert to as before.

claudey committed:
Stocks pages all responsive now.

claudey committed:
Change menu icon to correct image from Figma.

claudey committed:
Fixed tools content section to be repsonsive on mobile.

claudey committed:
Financial terms show page responsive now.

claudey committed:
AI section on home page responsive now.

claudey commented:
@Shpigford all pages have been optimized for mobile now.

Shpigford commented:
@claudey This looks amazing! I've got a few things to change/fix on my end, but otherwise is there anything else you wanted/needed to do on it?

claudey commented:
@Shpigford just a few gradients, really.

Shpigford committed:
Update contact email and improve meta titles for branding consistency

Shpigford committed:
Update branding language and improve call-to-action text for consistency across the site

Shpigford committed:
Merge branch 'main' into pr/301

Shpigford committed:
Refactor FeaturesController to validate categories and load assistant content; update view titles and descriptions for SEO

Shpigford committed:
Update brakeman.ignore to add dynamic render path warnings and bump Brakeman version to 7.0.2

Shpigford committed:
Update FAQ content and pricing page visuals for clarity and consistency. Replace outdated questions and answers in the FAQ section, and enhance the pricing page with updated imagery and call-to-action text.

Shpigford committed:
Remove "quick_links" partial from multiple stock views and update search placeholder text for clarity.

Shpigford committed:
Update pricing page call-to-action link and remove social share partial from shared views for a cleaner UI.

Shpigford committed:
Replace meta.png with meta.jpg to improve image quality and remove outdated format.

Shpigford committed:
Update multiple image assets for improved quality and remove outdated author image.

claudey committed:
Added HSL colors for home rainbow pattern.

claudey committed:
Appled styles from Rainbow Button as supplied by Justin.

claudey committed:
Extended Tailwind config to include colors, animation and keyframes.

Shpigford commented:
/tip $2500 @claudey

Shpigford commented:
@claudey Thank you so much for your work on this! Huge help to us and you did a fantastic job. We'll be deploying this later today!

claudey commented:
@Shpigford Thanks for your help in getting this out. I've been following Maybe for so long. I'm glad I got to contribute to this — it's special to me. Thanks for the tip too.

And @justinfar is an amazing designer!
